CMakeLists.txt

```
@@ -0,0 +1,31 @@
+cmake_minimum_required(VERSION 3.28)
+project(snitch)
+
+find_package(Eigen3 REQUIRED)
+find_package(MPI REQUIRED)
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_SOURCE_DIR}")
+find_package(LibMesh REQUIRED)
+
+include_directories(${EIGEN3_INCLUDE_DIR})
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/include)
+include_directories(${LIBMESH_INCLUDE_DIRS})
```

Recommend separating these between the external and the internal
```suggestion
include_directories(${LIBMESH_INCLUDE_DIRS})

include_directories(${CMAKE_CURRENT_SOURCE_DIR}/include)
```